docs/proposals/config_data.md

```diff
@@ -0,0 +1,319 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
+
+The main focus points of this proposal are:
+
+* Solving dynamic configuration problem of components.
+* Encapsulate configuration information to decrease component configuration complexity.
+* Create a flexible configuration model for ``Kubernetes`` components.
+* Increase overall system quality.
+
+## Motivation
+
+Currently command-line flags and environment variables are used for component configurations in ``Kubernetes``
+which is complicated and error-prone. This approach is not suitable for a distributed computing environment.
+Specifically the problems with the current approach are:
+
+* Complicated deployment orchestration
+* Hard to synchronize across replicated components.
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``ConfigData`` resource is proposed to solve the aforementioned problems
+that is stored in ``etcd`` and distributed to interested/necessary components
+through the API server. 
+
+The ``ConfigData`` architecture will be very similar to that of ``Secrets``.
+
+``ConfigData`` can either be mounted as a volume or can be injected as environment
+variables to a container through the use of the downward API.
+
+### Dynamic Configuration
+
+Any long-running system has mutating specification over time. In order to facilitate this functionality,
+``ConfigData`` will be versioned and updates will automatically made available to the container (downward API).
```

Comment:
Only the downward API volume plugin supports eventual consistency.  Are you suggesting that we will restart pods eventually that consume `ConfigData` via the downward API for env vars?  I assume so since the volume plugin is dealt with separately.  If so, I think it should be made more explicit what you're referring to.  Add a description and a note that we'll defer this to a follow up issue / proposal.